### DISCUSSION TITLE
How To Downloading WMTS/WMS Layers in GPKG Format

### DISCUSSION THREAD

**Discussion by gokboeruealp** [WorldWindEarth/WorldWindKotlin#79] [Q&A]
Hi,

I am able to retrieve maps from online WMTS or WMS servers. However, I need help figuring out how to download these maps, either in full or from a selected area, in GPKG format.

I think there is an example in this video: https://www.youtube.com/watch?v=UgkRhCw-nxc

I tried the following code but did not get the desired result:

```
wwd.mainScope.launch {
    val contentManager = GpkgContentManager(File(cacheDir, folderPath).absolutePath)
    val ogcLayerFactory = OgcLayerFactory(serviceAddress)
    val layer = ogcLayerFactory.createLayer(layerName)
    wwd.engine.layers.addLayer(layer.apply {
        configureCache(
            contentManager,
            layerName
        )
    })
}
```

If anyone can provide guidance or suggest a solution for this issue, I would greatly appreciate it. Thank you in advance!

**Comment by EMaksymenko** [comment]
You have configured offline cache i to gpkg, but then you need to call makeLocal method for some specified sector and resolution to start downloading it. By default, it saves everything what the user sees into cache.

**Comment by gokboeruealp** [comment]
thank you for your reply. 
my version: 1.5.7

I tried the makeLocal function but I did not get efficient results, what range should I use as resolution value?

**Reply by EMaksymenko** [reply]
Radians per pixel or depre per pixel.
